File: plugins/perl/test/test-sc.pl
# PerlScript test suite - ScriptCommand hook in perl
use vh;
use strict;

sub Say {
  my $msg = shift;
  $msg=~s/\&/&#36;/gmx;
  $msg=~s/\|/&#124;/gmx;
  vh::SendToClass($msg."|", 10, 10);
}

sub VH_OnScriptCommand {
  my $cmd = shift;
  my $data = shift;
  my $plug = shift;
  my $script = shift;
  Say "    (test-sc.pl:VH_OnScriptCommand) cmd=[$cmd] data=[$data] plug=[$plug] script=[$script]";
  return 1;
}

File: plugins/perl/test/test-sc.lua
-- PerlScript test suite - ScriptCommand hook in lua
function VH_OnScriptCommand(cmd, data, plug, script)
  VH:SendToClass("    (test-sc.lua:VH_OnScriptCommand) cmd=["..cmd.."] data=["..data.."] plug=["..plug.."] script=["..script.."]|", 10, 10)
  return 1
end
